codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/HttpBoundProtocolGenerator.kt

```diff
@@ -405,12 +405,19 @@ class HttpBoundProtocolGenerator(
         val contentType = httpBindingResolver.requestContentType(operationShape)
         httpBindingGenerator.renderUpdateHttpBuilder(implBlockWriter)
         httpBuilderFun(implBlockWriter) {
-            rust("let builder = #T::new();", RuntimeType.HttpRequestBuilder)
-            val additionalHeaders = listOf("Content-Type" to contentType) + protocol.additionalHeaders(operationShape)
+            Attribute.AllowUnusedMut.render(this)
```

Comment:
I don't think `AllowUnused` is required here—`content-type` additionalHeaders is always a non-empty list so we'll always have code that mutates the builder